src/config_file.ts

```diff
@@ -0,0 +1,216 @@
+    this.#content.deploy = normalizedArgs;
+  }
+
+  /**
+   * For every arg in `ConfigArgs`, if the `args` argument object does not contain
+   * the arg, fill it with the value in this `ConfigFile`, if any.
+   */
+  useAsDefaultFor(args: ConfigArgs) {
+    for (const [key, thisValue] of Object.entries(this.args())) {
+      // deno-lint-ignore no-explicit-any
+      if ((args as any)[key] === undefined && thisValue) {
+        // deno-lint-ignore no-explicit-any
+        (args as any)[key] = thisValue;
+      }
+    }
+  }
+
+  /** Check if the `ConfigArgs` in this `ConfigFile` match `args`
+   *
+   * Ignores any property in `args` not meant to be persisted.
+   */
+  eq(args: ConfigArgs) {
+    const otherConfigArgs = this.normalize(args);
+    // Iterate over the other args as they might include args not yet persisted in the config file
+    for (const [key, otherValue] of Object.entries(otherConfigArgs)) {
+      // deno-lint-ignore no-explicit-any
+      if ((this.args() as any)[key] !== otherValue) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  normalize(args: ConfigArgs): ConfigArgs {
+    // Copy object as normalization is internal to the config file
+    const normalizedArgs = {
+      project: args.project,
+      entrypoint: (args.entrypoint && !isURL(args.entrypoint))
+        ? resolve(args.entrypoint)
+        // Backoff if entrypoint is URL, the user knows what they're doing
+        : args.entrypoint,
+    };
+    return normalizedArgs;
+  }
+
+  /** Return whether the `ConfigFile` has the `deploy` namespace */
+  hasDeployConfig() {
+    return this.#content.deploy !== undefined;
+  }
+
+  static fromFileContent(filepath: string, content: string) {
+    const parsedContent = JSON.parse(content);
+    const configContent = {
+      ...parsedContent,
+      deploy: parsedContent.deploy && {
+        ...parsedContent.deploy,
+        entrypoint: parsedContent.deploy.entrypoint &&
+          (isURL(parsedContent.deploy.entrypoint)
+            // Backoff if entrypoint is URL, the user knows what they're doing
+            ? parsedContent.deploy.entrypoint
+            // entrypoint must be interpreted as absolute or relative to the config file
+            : resolve(dirname(filepath), parsedContent.deploy.entrypoint)),
+      },
+    };
+    return new ConfigFile(filepath, configContent);
+  }
+
+  toFileContent() {
+    const content = {
+      ...this.#content,
+      deploy: this.#content.deploy && {
+        ...this.#content.deploy,
+        entrypoint: this.#content.deploy.entrypoint &&
+          (isURL(this.#content.deploy.entrypoint)
+            // Backoff if entrypoint is URL, the user knows what they're doing
+            ? this.#content.deploy.entrypoint
+            // entrypoint must be stored relative to the config file
+            : relative(dirname(this.#path), this.#content.deploy.entrypoint)),
+      },
+    };
+    return JSON.stringify(content, null, 2);
```

Comment:
nit: I'm a little annoyed that this updates formatting for existing properties. For example:
```json
{
  "fmt": {
    "files": {
      "exclude": ["action/node_modules/"]
    }
  },
  "lint": {
    "files": {
      "exclude": ["action/node_modules/"]
    }
  },
  "tasks": {
    "test": "deno test -A --unstable tests/ src/",
    "build-action": "deno bundle ./src/utils/mod.ts > ./action/deps.js",
    "version-match": "deno run --allow-read --allow-env ./tools/version_match.ts"
  }
}
```
turned into

```json
{
  "fmt": {
    "files": {
      "exclude": [
        "action/node_modules/"
      ]
    }
  },
  "lint": {
    "files": {
      "exclude": [
        "action/node_modules/"
      ]
    }
  },
  "tasks": {
    "test": "deno test -A --unstable tests/ src/",
    "build-action": "deno bundle ./src/utils/mod.ts > ./action/deps.js",
    "version-match": "deno run --allow-read --allow-env ./tools/version_match.ts"
  },
  "deploy": {
    "project": "cool-dove",
    "entrypoint": "https://deno.com/examples/client_ip.js"
  }
}
```

`deno fmt` doesn't complain about both versions which I like.